dialdbg/src/rtt.rs

```diff
@@ -0,0 +1,28 @@
+use anyhow::{anyhow, Result};
+use std::{ops::Add, time};
+use viam::gen::proto::rpc::examples::echo::v1::{
+    echo_service_client::EchoServiceClient, EchoRequest,
+};
+use viam::rpc::dial::ViamChannel;
+
+// Returns the average round-trip-time over num_pings for the passed-in channel.
+pub(crate) async fn measure_rtt(ch: ViamChannel, num_pings: u32) -> Result<time::Duration> {
+    let mut total_ping = time::Duration::new(0, 0);
+    for _ in 0..num_pings {
+        let start = time::Instant::now();
+
+        // Send an echo request across the channel. It's unlikely the remote will be able to
+        // respond to this request, but we'll still get a good sense of RTT.
```

Comment:
I don't feel great about this strategy; I wish there were a request we could send across the channel that would never result in an error (notice the `ok()` call below to get rid of `Unimplemented` errors). In practice, this does seem to measure an RTT. To a rover in the office: ~120ms via gRPC and ~140ms via WebRTC. To a server on my machine: ~2ms via gRPC and ~1ms via WebRTC. Do let me know if you have suggestions for improvements here.

Reply:
I actually kind of think `echo` is good here. Ideally we want a no-op so we're only accounting for the actual network time and not for server-side processing time, but I don't think we can do that right now in rust. An unimplemented `echo` request seems like a decent enough proxy.